### DISCUSSION TITLE
Rate Limits while using PulseChain

### DISCUSSION THREAD

**Discussion by ryanhungate** [WalletConnect/walletconnect-monorepo#2418] [Developer Support] [answered]
Hey everyone, we had no problems at all while using the ETH chain, but we're getting reports now of being throttled while using PulseChain.com ( chain ID 369 ).

Is there anything that needs to be done to remove this limitation? We're barely scratching the surface right now for adoption and would be really bad if everyone got blocked. :)

Thanks!

**Comment by xzilja** [comment]
We actually don't support pulse chain on our rpc https://github.com/WalletConnect/web3modal/blob/V2/chains/ethereum/src/utils.ts#L15 so limitation is coming from another endpoint outside of web3modal (you are likely using default rpc from wagmi?).

**Comment by ryanhungate** [comment]
@0xAsimetriq yeah this is just us using the single `w3mProvider` option. Maybe we need to look into using another one as a fallback. I'll check into this and report back. 👍

**Reply by xzilja** [reply, marked as answer]
Yeh, in that case you will be using whichever rpc is defined in wagmi for pulse chain. It's likely a public one which has per ip limiter. If you hit it's limits, I'd recommend to set up wagmi's jsonRpc provider with a paid rpc of some sort.

**Comment by ryanhungate** [comment]
Sorry about that - I forgot to close this ;)
